src/controller/CHIPDevice.cpp

```diff
@@ -144,7 +144,9 @@ CHIP_ERROR Device::SendCommands()
     bool loadedSecureSession = false;
     ReturnErrorOnFailure(LoadSecureSessionParametersIfNeeded(loadedSecureSession));
     VerifyOrReturnError(mCommandSender != nullptr, CHIP_ERROR_INCORRECT_STATE);
-    return mCommandSender->SendCommandRequest(mDeviceId, mAdminId);
+    Messaging::ExchangeContext * exchange = mExchangeMgr->NewContext(mSecureSession, mCommandSender);
```

Comment:
This seems counter to how I thought this whole scheme works - controllers shouldn't have to pass in secure session handles to IM objects. Rather, they should be passing in a logical encapsulation of their destination, which could be a NodeId (in the case of CASE), or, a passcode (in the case of PASE) or, a group (which selects a group key).

The allocation of the exchange should then automatically figure out the 'right' secure session over which to allocate the exchange into.